pkg/query/iter.go

```diff
@@ -548,16 +554,18 @@ type dedupSeriesIterator struct {
 
 	penA, penB int64
 	useA       bool
+	hints      *storage.SelectHints
 }
 
-func newDedupSeriesIterator(a, b adjustableSeriesIterator) *dedupSeriesIterator {
+func newDedupSeriesIterator(a, b adjustableSeriesIterator, hints *storage.SelectHints) *dedupSeriesIterator {
```

Comment:
Let's avoid tieing low-level algorithm into some struct with many fields. We only care about step - which also is far from being precise.